datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -2712,4 +2757,75 @@ Projection: a, b
         \n    TableScan: test2";
         assert_optimized_plan_eq(&plan, expected)
     }
+
+    #[test]
+    fn test_push_down_volatile_function_in_aggregate() -> Result<()> {
+        // SELECT t.a, t.r FROM (SELECT a, SUM(b), random() AS r FROM test1 GROUP BY a) AS t WHERE t.a > 5 AND t.r > 0.5;
+        let table_scan = test_table_scan_with_name("test1")?;
+        let plan = LogicalPlanBuilder::from(table_scan)
+            .aggregate(vec![col("a")], vec![sum(col("b"))])?
+            .project(vec![col("a"), sum(col("b")), random().alias("r")])?
```

Comment:
I think it is important to test the expression like `random() + 1` rather than just `random()` for the reasons I listed above
```suggestion
            .project(vec![col("a"), sum(col("b")), random().add(lit(1)).alias("r")])?
```